ci/build_wheel.sh

```diff
@@ -33,9 +33,49 @@ rapids-generate-version > ./VERSION
 
 RAPIDS_PY_CUDA_SUFFIX="$(rapids-wheel-ctk-name-gen ${RAPIDS_CUDA_VERSION})"
 
+touch /tmp/constraints-build.txt
+
+if [[ "${package_name}" == "cuspatial" ]]; then
+  # Downloads libcuspatial wheel from this current build,
+  # then ensures 'cuspatial' wheel builds always use the 'libcuspatial' just built in the same CI run.
+  RAPIDS_PY_WHEEL_NAME="libcuspatial_${RAPIDS_PY_CUDA_SUFFIX}" rapids-download-wheels-from-s3 cpp /tmp/libcuspatial_dist
+  echo "libcuspatial-${RAPIDS_PY_CUDA_SUFFIX} @ file://$(echo /tmp/libcuspatial_dist/libcuspatial_*.whl)" > /tmp/constraints-build.txt
+fi
```

Comment:
It's not so important for cuspatial, but to establish a pattern that will translate better to all of the other repositories, what about putting the download back into `build_wheel_cuspatial.sh` and passing a constraint file through instead? Then this script would be responsible for creating an empty constraint file.

Alternatively, leave the logic here but have the script accept variables to indicate what packages need to be downloaded from s3 and added to constraints. Either way, we want a pattern that will support longer chains (e.g. libcugraph->pylibcugraph->cugraph).

Reply:
I've moved the download back in https://github.com/rapidsai/cuspatial/pull/1473/commits/29b005edb4d96a826ce8e6b03cabcb6ab5998124. 

That pattern is already supporting "longer chains" like `libcudf -> pylibcudf -> cudf` ([cudf code link](https://github.com/rapidsai/cudf/blob/768fbaa28033446dc899872b3c94213c75bc1a98/ci/build_wheel_cudf.sh#L10-L21)). Keeping that as-is will keep the diffs for these PRs smaller... I should have just done that from the beginning.

----

However ... I don't understand the suggestion "this script would be responsible for creating an empty constraint file" though... could you explain that?

`build_wheel_cuspatial.sh` is populated in a constraints file and populated environment variable `PIP_CONSTRAINT` with the path to it, so every `pip {command}` will respect that.

Other `build_wheel_{project}.sh` scripts are not setting that env variable, and so not using `pip`'s constraints mechanism.

What's the purpose of generating an empty constraints file?

Reply:
The version of the code here was already generating an empty constraints file by default. You had the `touch  /tmp/constraints-build.txt`, and then you replaced it in the `package_name == cuspatial` block. I assumed that you did this because in the libcuspatial build there were no constraints but you didn't want to modify the pip commands to include or exclude the constraint argument based on the name, so you chose to use an empty constraints file instead. Was I misunderstanding? I was attempting to give you a way to maintain that structure. Exporting the `PIP_CONSTRAINT` variable is cleaner though since then you always invoke pip without the `--constraint` argument. The other benefit of the env var is that it also works with build environment construction, although that is no longer relevant given the nature of this PR.

The point I was trying to make about longer chains is that it makes less sense to me to have the generic `build_wheel.sh` script have a bunch of `if package_name ==...` logic when you could instead have the `build_wheel_${package_name}.sh` have that logic and pass a standardized input to the `build_wheel.sh` script. Now that you've put it back I think that stays simpler.

Reply:
Ok I understand, thanks. Will leave it as-is (using `PIP_CONSTRAINT` env variable set in `ci/build_wheel_{project}.sh` instead of if-else blocks in `ci/build_wheel.sh`).